diff --git a/node_modules/@discord/bottom-sheet/src/components/bottomSheet/BottomSheet.tsx b/node_modules/@discord/bottom-sheet/src/components/bottomSheet/BottomSheet.tsx
index 2897fef..9a8505e 100644
--- a/node_modules/@discord/bottom-sheet/src/components/bottomSheet/BottomSheet.tsx
+++ b/node_modules/@discord/bottom-sheet/src/components/bottomSheet/BottomSheet.tsx
@@ -1382,7 +1382,8 @@ const BottomSheetComponent = forwardRef<BottomSheet, BottomSheetProps>(
           if (containerHeight !== _previousContainerHeight) {
             animationSource = ANIMATION_SOURCE.CONTAINER_RESIZE;
             animationConfig = {
-              duration: 0,
+              // https://github.com/gorhom/react-native-bottom-sheet/pull/1497
+              duration: 1,
             };
           }
         }
diff --git a/node_modules/@discord/bottom-sheet/src/components/bottomSheetHandleContainer/BottomSheetHandleContainer.tsx b/node_modules/@discord/bottom-sheet/src/components/bottomSheetHandleContainer/BottomSheetHandleContainer.tsx
index 2219e0f..59f90ba 100644
--- a/node_modules/@discord/bottom-sheet/src/components/bottomSheetHandleContainer/BottomSheetHandleContainer.tsx
+++ b/node_modules/@discord/bottom-sheet/src/components/bottomSheetHandleContainer/BottomSheetHandleContainer.tsx
@@ -92,10 +92,6 @@ function BottomSheetHandleContainerComponent({
     >
       <Animated.View
         key="BottomSheetHandleContainer"
-        accessible={true}
-        accessibilityRole="adjustable"
-        accessibilityLabel="Bottom Sheet handle"
-        accessibilityHint="Drag up or down to extend or minimize the Bottom Sheet"
         onLayout={handleContainerLayout}
       >
         <HandleComponent
diff --git a/node_modules/@discord/bottom-sheet/src/utilities/animate.ts b/node_modules/@discord/bottom-sheet/src/utilities/animate.ts
index 0ce4c9a..9562675 100644
--- a/node_modules/@discord/bottom-sheet/src/utilities/animate.ts
+++ b/node_modules/@discord/bottom-sheet/src/utilities/animate.ts
@@ -4,6 +4,7 @@ import {
   withTiming,
   withSpring,
   AnimationCallback,
+  ReduceMotion,
 } from 'react-native-reanimated';
 import { ANIMATION_CONFIGS, ANIMATION_METHOD } from '../constants';
 
@@ -26,6 +27,8 @@ export const animate = ({
     configs = ANIMATION_CONFIGS;
   }
 
+  configs.reduceMotion = ReduceMotion.Never;
+
   // detect animation type
   const type =
     'duration' in configs || 'easing' in configs
