diff --git a/node_modules/@envoy/react-native-key-commands/KeyCommands.podspec b/node_modules/@envoy/react-native-key-commands/KeyCommands.podspec
new file mode 100644
index 0000000..b0b8b4d
--- /dev/null
+++ b/node_modules/@envoy/react-native-key-commands/KeyCommands.podspec
@@ -0,0 +1,19 @@
+require 'json'
+
+package = JSON.parse(File.read(File.join(__dir__, 'package.json')))
+
+Pod::Spec.new do |s|
+  s.name         = "KeyCommands"
+  s.version      = package['version']
+  s.summary      = "iOS UIKeyCommand native component for React Native"
+  s.license      = package['license']
+
+  s.authors      = "Fang-Pen Lin <hello@fangpenlin.com> (https://fangpenlin.com)"
+  s.homepage     = "https://github.com/envoy/react-native-key-commands#readme"
+  s.platform     = :ios, "7.0"
+
+  s.source       = { :git => "https://github.com/envoy/react-native-key-commands.git" }
+  s.source_files  = "ios/**/*.{h,m}"
+
+  s.dependency 'React'
+end
\ No newline at end of file
diff --git a/node_modules/@envoy/react-native-key-commands/dist/index.js b/node_modules/@envoy/react-native-key-commands/dist/index.js
index 97acefa..9abe59a 100644
--- a/node_modules/@envoy/react-native-key-commands/dist/index.js
+++ b/node_modules/@envoy/react-native-key-commands/dist/index.js
@@ -7,7 +7,7 @@ var __rest = (this && this.__rest) || function (s, e) {
             t[p[i]] = s[p[i]];
     return t;
 };
-import { NativeModules, StyleSheet, requireNativeComponent } from 'react-native';
+import { NativeModules, StyleSheet, requireNativeComponent, Platform, View } from 'react-native';
 import React from 'react';
 const { KeyCommandConstants } = NativeModules;
 const RCTKeyCommands = requireNativeComponent('RCTKeyCommands');
@@ -16,21 +16,30 @@ const defaultStyles = StyleSheet.create({
         backgroundColor: 'transparent'
     }
 });
+const isAndroid = Platform.OS === 'android';
+
 export const constants = {
-    keyModifierShift: KeyCommandConstants.keyModifierShift,
-    keyModifierControl: KeyCommandConstants.keyModifierControl,
-    keyModifierAlternate: KeyCommandConstants.keyModifierAlternate,
-    keyModifierCommand: KeyCommandConstants.keyModifierCommand,
-    keyModifierNumericPad: KeyCommandConstants.keyModifierNumericPad,
-    keyInputUpArrow: KeyCommandConstants.keyInputUpArrow,
-    keyInputDownArrow: KeyCommandConstants.keyInputDownArrow,
-    keyInputLeftArrow: KeyCommandConstants.keyInputLeftArrow,
-    keyInputRightArrow: KeyCommandConstants.keyInputRightArrow,
-    keyInputEscape: KeyCommandConstants.keyInputEscape,
+    keyModifierShift: !isAndroid ? KeyCommandConstants.keyModifierShift : null,
+    keyModifierControl: !isAndroid ? KeyCommandConstants.keyModifierControl : null,
+    keyModifierAlternate: !isAndroid ? KeyCommandConstants.keyModifierAlternate : null,
+    keyModifierCommand: !isAndroid ? KeyCommandConstants.keyModifierCommand : null,
+    keyModifierNumericPad: !isAndroid ? KeyCommandConstants.keyModifierNumericPad : null,
+    keyInputUpArrow: !isAndroid ? KeyCommandConstants.keyInputUpArrow : null,
+    keyInputDownArrow: !isAndroid ? KeyCommandConstants.keyInputDownArrow : null,
+    keyInputLeftArrow: !isAndroid ? KeyCommandConstants.keyInputLeftArrow : null,
+    keyInputRightArrow: !isAndroid ? KeyCommandConstants.keyInputRightArrow : null,
+    keyInputEscape: !isAndroid ? KeyCommandConstants.keyInputEscape : null,
 };
 export default class KeyCommands extends React.Component {
     render() {
         const _a = this.props, { style } = _a, props = __rest(_a, ["style"]);
+        if (isAndroid) {
+            return (
+              <View
+                {...Object.assign({}, props, { style: [defaultStyles.main, style] })}
+              />
+            );
+        }
         return (<RCTKeyCommands {...Object.assign({}, props, { style: [defaultStyles.main, style] })}/>);
     }
 }
diff --git a/node_modules/@envoy/react-native-key-commands/src/index.tsx b/node_modules/@envoy/react-native-key-commands/src/index.tsx
index 0f287e4..de78d70 100644
--- a/node_modules/@envoy/react-native-key-commands/src/index.tsx
+++ b/node_modules/@envoy/react-native-key-commands/src/index.tsx
@@ -3,7 +3,9 @@ import {
   NativeSyntheticEvent,
   StyleSheet,
   ViewStyle,
-  requireNativeComponent
+  requireNativeComponent,
+  Platform,
+  View
 } from 'react-native'
 
 import React from 'react'
@@ -58,6 +60,13 @@ export interface Props {
 export default class KeyCommands extends React.Component<Props> {
   render () {
     const { style, ...props } = this.props
+    if (Platform.OS === 'android') {
+      return (
+        <View
+        {...{...props, style: [defaultStyles.main, style]}}
+        />
+      );
+    }
     return (
       <RCTKeyCommands
         {...{...props, style: [defaultStyles.main, style]}}
