diff --git a/node_modules/@rocket.chat/sdk/lib/drivers/ddp.ts b/node_modules/@rocket.chat/sdk/lib/drivers/ddp.ts
index d069a01..2296df0 100644
--- a/node_modules/@rocket.chat/sdk/lib/drivers/ddp.ts
+++ b/node_modules/@rocket.chat/sdk/lib/drivers/ddp.ts
@@ -68,7 +68,7 @@ export class Socket extends EventEmitter {
     this.config = {
       host: options.host || 'http://localhost:3000',
       useSsl: options.useSsl || false,
-      reopen: options.reopen || 10000,
+      reopen: options.reopen || 5000,
       ping: options.timeout || 10000
     }
 
