diff --git a/node_modules/@rocket.chat/message-parser/messageParser.js b/node_modules/@rocket.chat/message-parser/messageParser.js
index c1dc1bd..d25c5ed 100644
--- a/node_modules/@rocket.chat/message-parser/messageParser.js
+++ b/node_modules/@rocket.chat/message-parser/messageParser.js
@@ -3,5 +3,5 @@
 if (process.env.NODE_ENV === 'production') {
   module.exports = require('./dist/messageParser.production.js');
 } else {
-  module.exports = require('./dist/messageParser.development.js');
+  module.exports = require('./dist/messageParser.production.js');
 }
diff --git a/node_modules/@rocket.chat/message-parser/package.json b/node_modules/@rocket.chat/message-parser/package.json
index bb60b0d..d641865 100644
--- a/node_modules/@rocket.chat/message-parser/package.json
+++ b/node_modules/@rocket.chat/message-parser/package.json
@@ -19,9 +19,11 @@
   "main": "messageParser.js",
   "exports": {
     ".": {
+      "types": "./dist/index.d.ts",
       "default": "./messageParser.js"
     },
     "./index": {
+      "types": "./dist/index.d.ts",
       "default": "./messageParser.js"
     }
   },
