diff --git a/node_modules/@rocket.chat/sdk/lib/api/api.ts b/node_modules/@rocket.chat/sdk/lib/api/api.ts
index 0faaabd..a8bd048 100644
--- a/node_modules/@rocket.chat/sdk/lib/api/api.ts
+++ b/node_modules/@rocket.chat/sdk/lib/api/api.ts
@@ -160,10 +160,20 @@ class Client implements IClient {
     return { status, data }
 
   }
+
   private getParams (data: any) {
-    return Object.keys(data).map(function (k) {
-      return encodeURIComponent(k) + '=' + (typeof data[k] === 'object' ? encodeURIComponent(JSON.stringify(data[k])) : encodeURIComponent(data[k]))
-    }).join('&')
+    const params: any = [];
+    Object.keys(data).forEach(key => {
+      const value = data[key];
+      if (Array.isArray(value)) {
+        value.forEach(val => {
+          params.push(`${encodeURIComponent(key)}=${encodeURIComponent(val)}`);
+        });
+      } else {
+        params.push(`${encodeURIComponent(key)}=${encodeURIComponent(value)}`);
+      }
+    });
+    return params.join('&');
   }
 }
 
