diff --git a/node_modules/@rocket.chat/sdk/lib/api/api.ts b/node_modules/@rocket.chat/sdk/lib/api/api.ts
index 4f8228e..301432d 100644
--- a/node_modules/@rocket.chat/sdk/lib/api/api.ts
+++ b/node_modules/@rocket.chat/sdk/lib/api/api.ts
@@ -166,7 +166,7 @@ class Client implements IClient {
       const value = data[key];
       if (Array.isArray(value)) {
         value.forEach(val => {
-          params.push(`${encodeURIComponent(key)}=${encodeURIComponent(val)}`);
+          params.push(`${encodeURIComponent(key)}[]=${encodeURIComponent(val)}`);
         });
       } else {
         params.push(`${encodeURIComponent(key)}=${(typeof data[key] === 'object' ? encodeURIComponent(JSON.stringify(data[key])) : encodeURIComponent(data[key]))}`);
