diff --git a/node_modules/@unimodules/react-native-adapter/android/build.gradle b/node_modules/@unimodules/react-native-adapter/android/build.gradle
index 70d03a5..d17b348 100644
--- a/node_modules/@unimodules/react-native-adapter/android/build.gradle
+++ b/node_modules/@unimodules/react-native-adapter/android/build.gradle
@@ -66,8 +66,9 @@ dependencies {
   unimodule 'unimodules-permissions-interface'
   unimodule 'unimodules-image-loader-interface'
 
-  compileOnly('com.facebook.react:react-native:+') {
-    exclude group: 'com.android.support'
-  }
+  // compileOnly('com.facebook.react:react-native:+') {
+  //   exclude group: 'com.android.support'
+  // }
+  implementation 'com.facebook.react:react-native:+'
   api 'com.github.bumptech.glide:glide:4.9.0'
 }