diff --git a/node_modules/commonmark/lib/inlines.js b/node_modules/commonmark/lib/inlines.js
index 4179cfd..478bbd5 100644
--- a/node_modules/commonmark/lib/inlines.js
+++ b/node_modules/commonmark/lib/inlines.js
@@ -996,7 +996,7 @@ var parseEmail = function(block) {
     }
 }
 
-var reHashtag = XRegExp.cache('^#(\\pL[\\pL\\d\\-_.]*[\\pL\\d])');
+var reHashtag = XRegExp.cache('^#([\\pL\\d\\-_.]*[\\pL\\d])');
 var parseHashtag = function(block) {
     if (this.brackets) {
         // Don't perform autolinking while inside an explicit link
