diff --git a/node_modules/emoji-toolkit/lib/js/joypixels.js b/node_modules/emoji-toolkit/lib/js/joypixels.js
index 8e696e6..be03b51 100644
--- a/node_modules/emoji-toolkit/lib/js/joypixels.js
+++ b/node_modules/emoji-toolkit/lib/js/joypixels.js
@@ -111,6 +111,7 @@
         ':L':'1f615',
         '=L':'1f615',
         ':P':'1f61b',
+        ':p':'1f61b',
         '=P':'1f61b',
         ':b':'1f61b',
         ':O':'1f62e',
