diff --git a/node_modules/expo-permissions/android/build.gradle b/node_modules/expo-permissions/android/build.gradle
index 5d0fd9e..aa5ebf2 100644
--- a/node_modules/expo-permissions/android/build.gradle
+++ b/node_modules/expo-permissions/android/build.gradle
@@ -74,5 +74,5 @@ dependencies {
   unimodule 'unimodules-core'
   unimodule 'unimodules-permissions-interface'
 
-  api "com.android.support:appcompat-v7:${safeExtGet("supportLibVersion", "28.0.0")}"
+  api "androidx.appcompat:appcompat:1.0.0"
 }
diff --git a/node_modules/expo-permissions/android/src/main/java/expo/modules/permissions/PermissionsModule.java b/node_modules/expo-permissions/android/src/main/java/expo/modules/permissions/PermissionsModule.java
index 9027678..53edd85 100644
--- a/node_modules/expo-permissions/android/src/main/java/expo/modules/permissions/PermissionsModule.java
+++ b/node_modules/expo-permissions/android/src/main/java/expo/modules/permissions/PermissionsModule.java
@@ -12,7 +12,7 @@ import android.net.Uri;
 import android.os.Build;
 import android.os.Bundle;
 import android.provider.Settings;
-import android.support.v4.app.NotificationManagerCompat;
+import androidx.core.app.NotificationManagerCompat;
 
 import org.unimodules.core.ExportedModule;
 import org.unimodules.core.ModuleRegistry;
diff --git a/node_modules/expo-permissions/android/src/main/java/expo/modules/permissions/PermissionsService.java b/node_modules/expo-permissions/android/src/main/java/expo/modules/permissions/PermissionsService.java
index f9a3d06..b2d77cc 100644
--- a/node_modules/expo-permissions/android/src/main/java/expo/modules/permissions/PermissionsService.java
+++ b/node_modules/expo-permissions/android/src/main/java/expo/modules/permissions/PermissionsService.java
@@ -2,7 +2,7 @@ package expo.modules.permissions;
 
 import android.content.Context;
 import android.content.pm.PackageManager;
-import android.support.v4.content.ContextCompat;
+import androidx.core.content.ContextCompat;
 
 import java.util.Arrays;
 import java.util.Collections;
