diff --git a/node_modules/expo-constants/android/build.gradle b/node_modules/expo-constants/android/build.gradle
index 5afacdf..b1054a0 100644
--- a/node_modules/expo-constants/android/build.gradle
+++ b/node_modules/expo-constants/android/build.gradle
@@ -75,5 +75,5 @@ dependencies {
   unimodule 'unimodules-constants-interface'
 
   api 'com.facebook.device.yearclass:yearclass:2.1.0'
-  api "com.android.support:support-annotations:${safeExtGet("supportLibVersion", "28.0.0")}"
+  api "androidx.annotation:annotation:1.0.0"
 }
diff --git a/node_modules/expo-constants/android/src/main/java/expo/modules/constants/ConstantsService.java b/node_modules/expo-constants/android/src/main/java/expo/modules/constants/ConstantsService.java
index 4989672..888d84f 100644
--- a/node_modules/expo-constants/android/src/main/java/expo/modules/constants/ConstantsService.java
+++ b/node_modules/expo-constants/android/src/main/java/expo/modules/constants/ConstantsService.java
@@ -3,7 +3,7 @@ package expo.modules.constants;
 import android.content.Context;
 import android.content.res.Resources;
 import android.os.Build;
-import android.support.annotation.Nullable;
+import androidx.annotation.Nullable;
 import android.util.DisplayMetrics;
 
 import com.facebook.device.yearclass.YearClass;
