diff --git a/node_modules/expo-file-system/android/build.gradle b/node_modules/expo-file-system/android/build.gradle
index cd9ab67..83764fd 100644
--- a/node_modules/expo-file-system/android/build.gradle
+++ b/node_modules/expo-file-system/android/build.gradle
@@ -78,6 +78,6 @@ dependencies {
   api 'commons-io:commons-io:1.4'
   api 'com.squareup.okhttp3:okhttp:3.10.0'
   api 'com.squareup.okhttp3:okhttp-urlconnection:3.10.0'
-  api "com.android.support:support-v4:${safeExtGet("supportLibVersion", "28.0.0")}"
+  api "androidx.legacy:legacy-support-v4:1.0.0"
 }
   
diff --git a/node_modules/expo-file-system/android/src/main/AndroidManifest.xml b/node_modules/expo-file-system/android/src/main/AndroidManifest.xml
index da97401..ec3f033 100644
--- a/node_modules/expo-file-system/android/src/main/AndroidManifest.xml
+++ b/node_modules/expo-file-system/android/src/main/AndroidManifest.xml
@@ -9,7 +9,7 @@
     <application>
         <provider
             tools:replace="android:authorities"
-            android:name="android.support.v4.content.FileProvider"
+            android:name="androidx.core.content.FileProvider"
             android:authorities="${applicationId}.provider"
             android:exported="false"
             android:grantUriPermissions="true"
diff --git a/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java b/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java
index 93e4385..ee53f7c 100644
--- a/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java
+++ b/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java
@@ -6,7 +6,7 @@ import android.content.res.Resources;
 import android.net.Uri;
 import android.os.AsyncTask;
 import android.os.Bundle;
-import android.support.v4.content.FileProvider;
+import androidx.core.content.FileProvider;
 import android.util.Base64;
 import android.util.Log;
 
