diff --git a/node_modules/expo-asset/android/src/main/java/expo/modules/asset/AssetModule.kt b/node_modules/expo-asset/android/src/main/java/expo/modules/asset/AssetModule.kt
index 9f1e2c6..61a1032 100644
--- a/node_modules/expo-asset/android/src/main/java/expo/modules/asset/AssetModule.kt
+++ b/node_modules/expo-asset/android/src/main/java/expo/modules/asset/AssetModule.kt
@@ -80,6 +80,12 @@ class AssetModule : Module() {
     Name("ExpoAsset")
 
     AsyncFunction("downloadAsync") Coroutine { uri: URI, md5Hash: String?, type: String ->
+      // If the scheme is null, it's a raw asset (e.g., "custom.ttf").
+      // Returning early prevents Expo from trying to "cache" a local native file.
+      if (uri.scheme == null) {
+          return@Coroutine uri
+      }
+
       if (uri.scheme == "file" && !uri.toString().startsWith(ANDROID_EMBEDDED_URL_BASE_RESOURCE)) {
         return@Coroutine uri
       }
