diff --git a/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/legacy/FileSystemLegacyModule.kt b/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/legacy/FileSystemLegacyModule.kt
--- a/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/legacy/FileSystemLegacyModule.kt
+++ b/node_modules/expo-file-system/android/src/main/java/expo/modules/filesystem/legacy/FileSystemLegacyModule.kt
@@ -84,11 +84,20 @@
     get() = appContext.persistentFilesDirectory
   private val cacheDirectory: File
     get() = appContext.cacheDirectory
-  private var client: OkHttpClient? = null
   private var dirPermissionsRequest: Promise? = null
   private val taskHandlers: MutableMap<String, TaskHandler> = HashMap()
   private val moduleCoroutineScope = CoroutineScope(Dispatchers.Default)
 
+  companion object {
+    @JvmStatic
+    var client: OkHttpClient? = null
+
+    @JvmStatic
+    fun setOkHttpClient(okHttpClient: OkHttpClient) {
+      client = okHttpClient
+    }
+  }
+
   @SuppressLint("WrongConstant", "DiscouragedApi")
   override fun definition() = ModuleDefinition {
     Name("ExponentFileSystem")
