diff --git a/node_modules/react-native-jitsi-meet/android/build.gradle b/node_modules/react-native-jitsi-meet/android/build.gradle
index 1efa23d..ed1e4cc 100644
--- a/node_modules/react-native-jitsi-meet/android/build.gradle
+++ b/node_modules/react-native-jitsi-meet/android/build.gradle
@@ -45,7 +45,11 @@ repositories {
 }
 
 dependencies {
-    implementation ('org.jitsi.react:jitsi-meet-sdk:2.3.0') {
+    implementation ('org.jitsi.react:jitsi-meet-sdk:2.3.1') {
+      exclude group: 'com.facebook.react', module:'react-native-fast-image'
+      exclude group: 'com.facebook.react', module:'react-native-vector-icons'
+      exclude group: 'com.facebook.react', module:'react-native-webview'
+      exclude group: 'com.facebook.react', module:'react-native-background-timer'
       transitive = true
     }
 }
