diff --git a/node_modules/react-native-scroll-bottom-sheet/src/index.tsx b/node_modules/react-native-scroll-bottom-sheet/src/index.tsx
index 7d4a54d..155d181 100644
--- a/node_modules/react-native-scroll-bottom-sheet/src/index.tsx
+++ b/node_modules/react-native-scroll-bottom-sheet/src/index.tsx
@@ -10,6 +10,8 @@ import {
   SectionListProps,
   StyleSheet,
   View,
+  StyleProp,
+  ViewStyle,
 } from 'react-native';
 import Animated, {
   abs,
@@ -156,6 +158,10 @@ type CommonProps = {
    * Reference to FlatList, ScrollView or SectionList in order to execute its imperative methods.
    */
   innerRef: RefObject<FlatList | ScrollView | SectionList>;
+  /*
+   * Style to be applied handling the handle and the content.
+   */
+  containerStyle: StyleProp<ViewStyle>;
 };
 
 type Props<T> = CommonProps &
@@ -595,6 +601,7 @@ export class ScrollBottomSheet<T extends any> extends Component<Props<T>> {
       componentType,
       onSettle,
       animatedPosition,
+      containerStyle,
       ...rest
     } = this.props;
     const AnimatedScrollableComponent = this.scrollComponent;
@@ -606,6 +613,7 @@ export class ScrollBottomSheet<T extends any> extends Component<Props<T>> {
         style={[
           StyleSheet.absoluteFillObject,
           // @ts-ignore
+          containerStyle,
           {
             transform: [{ translateY: this.translateY }],
           },
