diff --git a/node_modules/react-native-jitsi-meet/android/build.gradle b/node_modules/react-native-jitsi-meet/android/build.gradle
index 369b894..7fc14e2 100644
--- a/node_modules/react-native-jitsi-meet/android/build.gradle
+++ b/node_modules/react-native-jitsi-meet/android/build.gradle
@@ -36,9 +36,10 @@ android {
 }
 
 repositories {
-  maven {
-      url "https://github.com/jitsi/jitsi-maven-repository/raw/master/releases"
-  }
+  // maven {
+  //     url "https://github.com/jitsi/jitsi-maven-repository/raw/master/releases"
+  // }
+  maven { url "file:/tmp/repo" }
   google()
   mavenCentral()
   jcenter()
@@ -46,7 +47,7 @@ repositories {
 
 dependencies {
     //resolutionStrategy.force "org.webkit:android-jsc:r224109"
-    implementation ('org.jitsi.react:jitsi-meet-sdk:2.1.0') {
+    implementation ('org.jitsi.react:jitsi-meet-sdk:+') {
       transitive = true
       //compile ("org.webkit:android-jsc:r224109") { force = true }
     }
diff --git a/node_modules/react-native-jitsi-meet/android/src/main/java/com/reactnativejitsimeet/JitsiMeetModule.java b/node_modules/react-native-jitsi-meet/android/src/main/java/com/reactnativejitsimeet/JitsiMeetModule.java
index e966745..297649b 100644
--- a/node_modules/react-native-jitsi-meet/android/src/main/java/com/reactnativejitsimeet/JitsiMeetModule.java
+++ b/node_modules/react-native-jitsi-meet/android/src/main/java/com/reactnativejitsimeet/JitsiMeetModule.java
@@ -8,6 +8,7 @@ import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactContextBaseJavaModule;
 import com.facebook.react.bridge.ReactMethod;
 import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.bridge.ReadableMap;
 import com.facebook.react.modules.core.DeviceEventManagerModule;
 
 public class JitsiMeetModule extends ReactContextBaseJavaModule {
@@ -40,10 +41,12 @@ public class JitsiMeetModule extends ReactContextBaseJavaModule {
     }
 
     @ReactMethod
-    public void call(String url) {
+    public void call(String url, ReadableMap options) {
         ReactApplicationContext context = getReactApplicationContext();
         Intent intent = new Intent(context, JitsiMeetNavigatorActivity.class);
+        Boolean videoMuted = options.hasKey("videoMuted") ? options.getBoolean("videoMuted") : false;
         intent.putExtra("url", url);
+        intent.putExtra("videoMuted", videoMuted);
         intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
         context.startActivity(intent);
     }
diff --git a/node_modules/react-native-jitsi-meet/android/src/main/java/com/reactnativejitsimeet/JitsiMeetNavigatorActivity.java b/node_modules/react-native-jitsi-meet/android/src/main/java/com/reactnativejitsimeet/JitsiMeetNavigatorActivity.java
index 3855488..b258ffd 100644
--- a/node_modules/react-native-jitsi-meet/android/src/main/java/com/reactnativejitsimeet/JitsiMeetNavigatorActivity.java
+++ b/node_modules/react-native-jitsi-meet/android/src/main/java/com/reactnativejitsimeet/JitsiMeetNavigatorActivity.java
@@ -9,7 +9,7 @@ import android.os.Bundle;
 import android.provider.Settings;
 import android.util.Log;
 import android.content.Intent;
-import android.support.v4.app.FragmentActivity;
+import androidx.fragment.app.FragmentActivity;
 
 import com.facebook.react.ReactInstanceManager;
 import com.facebook.react.modules.core.PermissionListener;
@@ -50,10 +50,12 @@ public class JitsiMeetNavigatorActivity extends FragmentActivity implements Jits
         super.onCreate(savedInstanceState);
 
         String url = getIntent().getStringExtra("url");
+        Boolean videoMuted = getIntent().getExtras().getBoolean("videoMuted");
         view = new JitsiMeetView(this);
         view.setListener(this);
         JitsiMeetConferenceOptions options = new JitsiMeetConferenceOptions.Builder()
             .setRoom(url)
+            .setVideoMuted(videoMuted)
             .build();
         view.join(options);
 
diff --git a/node_modules/react-native-jitsi-meet/ios/JitsiMeetViewController.h b/node_modules/react-native-jitsi-meet/ios/JitsiMeetViewController.h
index b68acd5..2a1e05a 100644
--- a/node_modules/react-native-jitsi-meet/ios/JitsiMeetViewController.h
+++ b/node_modules/react-native-jitsi-meet/ios/JitsiMeetViewController.h
@@ -13,6 +13,7 @@
 
 - (void) setDelegate:(id<JitsiMeetViewDelegate>) delegate;
 - (void)loadUrl:(NSString *) url;
+- (void)call:(NSString *)url options:(NSDictionary *)options;
 
 @end
 
diff --git a/node_modules/react-native-jitsi-meet/ios/JitsiMeetViewController.m b/node_modules/react-native-jitsi-meet/ios/JitsiMeetViewController.m
index 005824a..33a2187 100644
--- a/node_modules/react-native-jitsi-meet/ios/JitsiMeetViewController.m
+++ b/node_modules/react-native-jitsi-meet/ios/JitsiMeetViewController.m
@@ -21,6 +21,22 @@ - (void) setDelegate:(id<JitsiMeetViewDelegate>) delegate {
     }
 }
 
+- (BOOL)checkKey:(NSString *)key options:(NSDictionary *)options
+{
+  id obj = options[key];
+  if (obj != nil) {
+    if (obj != (id)[NSNull null]) {
+      return [[options objectForKey:key] boolValue];
+    }
+    else {
+      return NO;
+    }
+  }
+  else {
+    return NO;
+  }
+}
+
 - (void)loadUrl:(NSString *) url {
     JitsiMeetView *jitsiMeetView = (JitsiMeetView *) self.view;
     JitsiMeetConferenceOptions *options = [JitsiMeetConferenceOptions fromBuilder:^(JitsiMeetConferenceOptionsBuilder *builder) {        builder.room = url;
@@ -28,6 +44,15 @@ - (void)loadUrl:(NSString *) url {
     [jitsiMeetView join:options];
 }
 
+- (void)call:(NSString *)url options:(NSDictionary *)options {
+  JitsiMeetView *jitsiMeetView = (JitsiMeetView *) self.view;
+  JitsiMeetConferenceOptions *optionsBuilder = [JitsiMeetConferenceOptions fromBuilder:^(JitsiMeetConferenceOptionsBuilder *builder) {
+    builder.room = url;
+    builder.videoMuted = [self checkKey:@"videoMuted" options:options];
+  }];
+  [jitsiMeetView join:optionsBuilder];
+}
+
 
 - (void)didReceiveMemoryWarning {
     [super didReceiveMemoryWarning];
diff --git a/node_modules/react-native-jitsi-meet/ios/RNJitsiMeetNavigator.m b/node_modules/react-native-jitsi-meet/ios/RNJitsiMeetNavigator.m
index e641838..d579077 100644
--- a/node_modules/react-native-jitsi-meet/ios/RNJitsiMeetNavigator.m
+++ b/node_modules/react-native-jitsi-meet/ios/RNJitsiMeetNavigator.m
@@ -41,15 +41,15 @@ - (void)enterPictureInPicture:(NSDictionary *)data {
     jitsiMeetViewController = [storyboard instantiateViewControllerWithIdentifier:@"jitsiMeetStoryBoardID"];
 }
 
-RCT_EXPORT_METHOD(call:(NSString *)urlString)
+RCT_EXPORT_METHOD(call:(NSString *)urlString options:(NSDictionary *)options)
 {
-    RCTLogInfo(@"Load URL %@", urlString);
+    RCTLogInfo(@"Conference on %@ with Options %@", urlString, options);
     dispatch_sync(dispatch_get_main_queue(), ^{
         UIViewController* rootViewController = [[[[UIApplication sharedApplication]delegate] window] rootViewController];
         UINavigationController *navigationController = (UINavigationController *) rootViewController;
         [navigationController pushViewController:jitsiMeetViewController animated:true];
         [jitsiMeetViewController setDelegate:self];
-        [jitsiMeetViewController loadUrl:urlString];
+        [jitsiMeetViewController call:urlString options:options];
     });
 }
 
