diff --git a/node_modules/react-native-restart/android/build.gradle b/node_modules/react-native-restart/android/build.gradle
index 7597ac6..47a5d21 100644
--- a/node_modules/react-native-restart/android/build.gradle
+++ b/node_modules/react-native-restart/android/build.gradle
@@ -5,7 +5,7 @@ buildscript {
   if (project == rootProject) {
     repositories {
       google()
-      jcenter()
+      mavenCentral()
     }
 
     dependencies {
@@ -45,7 +45,6 @@ android {
 
 repositories {
   mavenCentral()
-  jcenter()
   google()
 
   def found = false
