diff --git a/node_modules/react-native/Libraries/Network/RCTHTTPRequestHandler.mm b/node_modules/react-native/Libraries/Network/RCTHTTPRequestHandler.mm
index b7b2db1..62c4c7a 100644
--- a/node_modules/react-native/Libraries/Network/RCTHTTPRequestHandler.mm
+++ b/node_modules/react-native/Libraries/Network/RCTHTTPRequestHandler.mm
@@ -85,6 +85,8 @@ RCT_EXPORT_MODULE()
       configuration = urlSessionConfigurationProvider();
     } else {
       configuration = [NSURLSessionConfiguration defaultSessionConfiguration];
+      configuration.HTTPAdditionalHeaders = @{ @"Connection": @"close" };
+
       // Set allowsCellularAccess to NO ONLY if key ReactNetworkForceWifiOnly exists AND its value is YES
       if (useWifiOnly) {
         configuration.allowsCellularAccess = ![useWifiOnly boolValue];
