diff --git a/node_modules/react-native-webview/android/.project b/node_modules/react-native-webview/android/.project
new file mode 100644
index 0000000..122755b
--- /dev/null
+++ b/node_modules/react-native-webview/android/.project
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>react-native-webview</name>
+	<comment>Project react-native-webview created by Buildship.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+	<filteredResources>
+		<filter>
+			<id>1760118463063</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|\.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>
diff --git a/node_modules/react-native-webview/apple/RNCWebViewImpl.m b/node_modules/react-native-webview/apple/RNCWebViewImpl.mm
similarity index 99%
rename from node_modules/react-native-webview/apple/RNCWebViewImpl.m
rename to node_modules/react-native-webview/apple/RNCWebViewImpl.mm
index 7f5c24d..0627f64 100644
--- a/node_modules/react-native-webview/apple/RNCWebViewImpl.m
+++ b/node_modules/react-native-webview/apple/RNCWebViewImpl.mm
@@ -917,7 +917,7 @@ RCTAutoInsetsProtocol>
 
   if(subview == nil) return;
 
-  Class class = subview.class;
+  Class viewClass = subview.class;
 
   NSOperatingSystemVersion iOS_11_3_0 = (NSOperatingSystemVersion){11, 3, 0};
   NSOperatingSystemVersion iOS_12_2_0 = (NSOperatingSystemVersion){12, 2, 0};
