diff --git a/node_modules/react-native-animatable/registry.js b/node_modules/react-native-animatable/registry.js
index c1f3a5e..ac9e6fb 100644
--- a/node_modules/react-native-animatable/registry.js
+++ b/node_modules/react-native-animatable/registry.js
@@ -15,10 +15,9 @@ export function getAnimationNames() {
 }
 
 export function initializeRegistryWithDefinitions(definitions) {
-  Object.keys(definitions).forEach(animationName => {
-    registerAnimation(
-      animationName,
-      createAnimation(definitions[animationName]),
-    );
-  });
+	Object.keys(definitions).forEach(animationName => {
+		if (typeof definitions[animationName] === 'object') {
+			registerAnimation(animationName, createAnimation(definitions[animationName]));
+		}
+	});
 }
