diff --git a/node_modules/react-native-webview/android/.project b/node_modules/react-native-webview/android/.project
new file mode 100644
index 0000000..122755b
--- /dev/null
+++ b/node_modules/react-native-webview/android/.project
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>react-native-webview</name>
+	<comment>Project react-native-webview created by Buildship.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+	<filteredResources>
+		<filter>
+			<id>1760118463063</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|\.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>
diff --git a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java
index 80c6803..4c6fff6 100644
--- a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java
+++ b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java
@@ -77,6 +77,7 @@ public class RNCWebView extends WebView implements LifecycleEventListener {
     private OnScrollDispatchHelper mOnScrollDispatchHelper;
     protected boolean hasScrollEvent = false;
     protected boolean nestedScrollEnabled = false;
+    public boolean ignoreSslErrors = false;
     protected ProgressChangedFilter progressChangedFilter;
 
     /**
@@ -219,6 +220,7 @@ public class RNCWebView extends WebView implements LifecycleEventListener {
     public void setWebViewClient(WebViewClient client) {
         super.setWebViewClient(client);
         if (client instanceof RNCWebViewClient) {
+            ((RNCWebViewClient) client).setReactContext(this.getReactApplicationContext());
             mRNCWebViewClient = (RNCWebViewClient) client;
             mRNCWebViewClient.setProgressChangedFilter(progressChangedFilter);
         }
diff --git a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewClient.java b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewClient.java
index 0287e1c..a44ddf4 100644
--- a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewClient.java
+++ b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewClient.java
@@ -35,6 +35,11 @@ import com.reactnativecommunity.webview.events.TopShouldStartLoadWithRequestEven
 import android.webkit.CookieManager;
 import android.webkit.CookieSyncManager;
 
+import java.security.cert.X509Certificate;
+import java.security.PrivateKey;
+import android.webkit.ClientCertRequest;
+import android.os.AsyncTask;
+import android.security.KeyChain;
 import java.util.concurrent.atomic.AtomicReference;
 
 public class RNCWebViewClient extends WebViewClient {
@@ -45,10 +50,60 @@ public class RNCWebViewClient extends WebViewClient {
     protected RNCWebView.ProgressChangedFilter progressChangedFilter = null;
     protected @Nullable RNCBasicAuthCredential basicAuthCredential = null;
 
+    private static String certificateAlias = null;
+    private ReactContext reactContext;
+
     public void setBasicAuthCredential(@Nullable RNCBasicAuthCredential credential) {
         basicAuthCredential = credential;
     }
 
+    public static void setCertificateAlias(String alias) {
+        certificateAlias = alias;
+    }
+
+    public void setReactContext(ReactContext reactContext) {
+        this.reactContext = reactContext;
+    }
+
+    @Override
+    public void onReceivedClientCertRequest(WebView view, ClientCertRequest request) {
+        class SslStuff {
+            PrivateKey privKey;
+            X509Certificate[] certChain;
+
+            public SslStuff(PrivateKey privKey, X509Certificate[] certChain) {
+                this.privKey = privKey;
+                this.certChain = certChain;
+            }
+        }
+
+        if (certificateAlias != null && reactContext != null) {
+            AsyncTask<Void, Void, SslStuff> task = new AsyncTask<Void, Void, SslStuff>() {
+                @Override
+                protected SslStuff doInBackground(Void... params) {
+                    try {
+                        PrivateKey privKey = KeyChain.getPrivateKey(reactContext, certificateAlias);
+                        X509Certificate[] certChain = KeyChain.getCertificateChain(reactContext, certificateAlias);
+
+                        return new SslStuff(privKey, certChain);
+                    } catch (Exception e) {
+                        return null;
+                    }
+                }
+
+                @Override
+                protected void onPostExecute(SslStuff sslStuff) {
+                    if (sslStuff != null) {
+                        request.proceed(sslStuff.privKey, sslStuff.certChain);
+                    }
+                }
+            };
+            task.execute();
+        } else {
+            super.onReceivedClientCertRequest(view, request);
+        }
+    }
+
     @Override
     public void onPageFinished(WebView webView, String url) {
         super.onPageFinished(webView, url);
@@ -157,6 +212,10 @@ public class RNCWebViewClient extends WebViewClient {
 
     @Override
     public void onReceivedSslError(final WebView webView, final SslErrorHandler handler, final SslError error) {
+        if (webView instanceof RNCWebView && ((RNCWebView) webView).ignoreSslErrors) {
+            handler.proceed();
+            return;
+        }
         // onReceivedSslError is called for most requests, per Android docs: https://developer.android.com/reference/android/webkit/WebViewClient#onReceivedSslError(android.webkit.WebView,%2520android.webkit.SslErrorHandler,%2520android.net.http.SslError)
         // WebView.getUrl() will return the top-level window URL.
         // If a top-level navigation triggers this error handler, the top-level URL will be the failing URL (not the URL of the currently-rendered page).
diff --git a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManagerImpl.kt b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManagerImpl.kt
index f83033b..8a98930 100644
--- a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManagerImpl.kt
+++ b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManagerImpl.kt
@@ -637,6 +637,10 @@ class RNCWebViewManagerImpl(private val newArch: Boolean = false) {
         view.settings.minimumFontSize = value
     }
 
+    fun setIgnoreSslErrors(viewWrapper: RNCWebViewWrapper, value: Boolean) {
+        viewWrapper.webView.ignoreSslErrors = value
+    }
+
     fun setAllowsProtectedMedia(viewWrapper: RNCWebViewWrapper, enabled: Boolean) {
       val view = viewWrapper.webView
       // This variable is used to keep consistency
diff --git a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModuleImpl.java b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModuleImpl.java
index 951749c..05792b8 100644
--- a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModuleImpl.java
+++ b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModuleImpl.java
@@ -551,4 +551,8 @@ public class RNCWebViewModuleImpl implements ActivityEventListener {
         }
         return (PermissionAwareActivity) activity;
     }
+
+    public static void setCertificateAlias(String alias) {
+        RNCWebViewClient.setCertificateAlias(alias);
+    }
 }
\ No newline at end of file
diff --git a/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewManager.java b/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewManager.java
index 4709e28..86696fe 100644
--- a/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewManager.java
+++ b/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewManager.java
@@ -45,6 +45,10 @@ public class RNCWebViewManager extends ViewGroupManager<RNCWebViewWrapper>
         mRNCWebViewManagerImpl = new RNCWebViewManagerImpl(true);
     }
 
+    public static void setCertificateAlias(String alias) {
+        RNCWebViewModule.setCertificateAlias(alias);
+    }
+
     @Nullable
     @Override
     protected ViewManagerDelegate<RNCWebViewWrapper> getDelegate() {
@@ -88,6 +92,12 @@ public class RNCWebViewManager extends ViewGroupManager<RNCWebViewWrapper>
         mRNCWebViewManagerImpl.setAllowsFullscreenVideo(view, value);
     }
 
+    @Override
+    @ReactProp(name = "ignoreSslErrors")
+    public void setIgnoreSslErrors(RNCWebViewWrapper view, boolean value) {
+        mRNCWebViewManagerImpl.setIgnoreSslErrors(view, value);
+    }
+
     @Override
     @ReactProp(name = "allowsProtectedMedia")
     public void setAllowsProtectedMedia(RNCWebViewWrapper view, boolean value) {
diff --git a/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewModule.java b/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewModule.java
index 4df0aae..867aa73 100644
--- a/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewModule.java
+++ b/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewModule.java
@@ -49,6 +49,10 @@ public class RNCWebViewModule extends NativeRNCWebViewModuleSpec {
         return mRNCWebViewModuleImpl.grantFileDownloaderPermissions(downloadingMessage, lackPermissionToDownloadMessage);
     }
 
+    public static void setCertificateAlias(String alias) {
+        RNCWebViewModuleImpl.setCertificateAlias(alias);
+    }
+
     @NonNull
     @Override
     public String getName() {
diff --git a/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewManager.java b/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewManager.java
index 974337f..d5814ab 100644
--- a/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewManager.java
+++ b/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewManager.java
@@ -32,6 +32,10 @@ public class RNCWebViewManager extends ViewGroupManager<RNCWebViewWrapper> {
         mRNCWebViewManagerImpl = new RNCWebViewManagerImpl();
     }
 
+    public static void setCertificateAlias(String alias) {
+        RNCWebViewModule.setCertificateAlias(alias);
+    }
+
     @Override
     public String getName() {
         return RNCWebViewManagerImpl.NAME;
@@ -67,6 +71,11 @@ public class RNCWebViewManager extends ViewGroupManager<RNCWebViewWrapper> {
         mRNCWebViewManagerImpl.setAllowsFullscreenVideo(view, value);
     }
 
+    @ReactProp(name = "ignoreSslErrors")
+    public void setIgnoreSslErrors(RNCWebViewWrapper view, boolean value) {
+        mRNCWebViewManagerImpl.setIgnoreSslErrors(view, value);
+    }
+
     @ReactProp(name = "allowsProtectedMedia")
     public void setAllowsProtectedMedia(RNCWebViewWrapper view, boolean value) {
         mRNCWebViewManagerImpl.setAllowsProtectedMedia(view, value);
diff --git a/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewModule.java b/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewModule.java
index 526cc16..8f3318d 100644
--- a/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewModule.java
+++ b/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewModule.java
@@ -51,6 +51,10 @@ public class RNCWebViewModule extends ReactContextBaseJavaModule {
         return mRNCWebViewModuleImpl.grantFileDownloaderPermissions(downloadingMessage, lackPermissionToDownloadMessage);
     }
 
+    public static void setCertificateAlias(String alias) {
+        RNCWebViewModuleImpl.setCertificateAlias(alias);
+    }
+
     @NonNull
     @Override
     public String getName() {
diff --git a/node_modules/react-native-webview/src/RNCWebViewNativeComponent.ts b/node_modules/react-native-webview/src/RNCWebViewNativeComponent.ts
index 7f4e6c4..ab1db2f 100644
--- a/node_modules/react-native-webview/src/RNCWebViewNativeComponent.ts
+++ b/node_modules/react-native-webview/src/RNCWebViewNativeComponent.ts
@@ -143,6 +143,7 @@ type WebViewDownloadEvent = Readonly<{
 // type MenuItem = Readonly<{label: string, key: string}>;
 
 export interface NativeProps extends ViewProps {
+  ignoreSslErrors?: boolean;
   // Android only
   allowFileAccess?: boolean;
   allowsProtectedMedia?: boolean;
diff --git a/node_modules/react-native-webview/src/WebViewTypes.ts b/node_modules/react-native-webview/src/WebViewTypes.ts
index eca8141..1ca3f17 100644
--- a/node_modules/react-native-webview/src/WebViewTypes.ts
+++ b/node_modules/react-native-webview/src/WebViewTypes.ts
@@ -1171,6 +1171,11 @@ export interface AndroidWebViewProps extends WebViewSharedProps {
 }
 
 export interface WebViewSharedProps extends ViewProps {
+  /**
+   * Accept invalid or self-signed TLS certificates. Use only for trusted hosts.
+   */
+  ignoreSslErrors?: boolean;
+
   /**
    * Loads static html or a uri (with optional headers) in the WebView.
    */
