diff --git a/node_modules/react-native-markdown-renderer/src/index.js b/node_modules/react-native-markdown-renderer/src/index.js
index 653bba2..e5cb521 100644
--- a/node_modules/react-native-markdown-renderer/src/index.js
+++ b/node_modules/react-native-markdown-renderer/src/index.js
@@ -88,9 +88,15 @@ export default class Markdown extends Component {
     }),
   };
 
-  copy = '';
-  renderer = null;
-  markdownParser = null;
+  constructor(props) {
+    super(props);
+    this.copy = '';
+    this.renderer = null;
+    this.markdownParser = null;
+  }
+  // copy = '';
+  // renderer = null;
+  // markdownParser = null;
 
   /**
    * Only when the copy changes will the markdown render again.
