diff --git a/node_modules/react-native-ui-lib/lib/ios/reactnativeuilib/keyboardtrackingview/ObservingInputAccessoryViewTemp.m b/node_modules/react-native-ui-lib/lib/ios/reactnativeuilib/keyboardtrackingview/ObservingInputAccessoryViewTemp.m
index 1ca52e8..323a04c 100644
--- a/node_modules/react-native-ui-lib/lib/ios/reactnativeuilib/keyboardtrackingview/ObservingInputAccessoryViewTemp.m
+++ b/node_modules/react-native-ui-lib/lib/ios/reactnativeuilib/keyboardtrackingview/ObservingInputAccessoryViewTemp.m
@@ -115,21 +115,25 @@ - (void)setHeight:(CGFloat)height
 
 - (void)_keyboardWillShowNotification:(NSNotification*)notification
 {
-    _keyboardState = KeyboardStateWillShow;
+    if (_keyboardState != KeyboardStateShown) {
+        _keyboardState = KeyboardStateWillShow;
         
-    [self invalidateIntrinsicContentSize];
+        [self invalidateIntrinsicContentSize];
         
-    if([_delegate respondsToSelector:@selector(ObservingInputAccessoryViewTempKeyboardWillAppear:keyboardDelta:)])
-    {
-        [_delegate ObservingInputAccessoryViewTempKeyboardWillAppear:self keyboardDelta:_keyboardHeight - _previousKeyboardHeight];
+        if([_delegate respondsToSelector:@selector(ObservingInputAccessoryViewTempKeyboardWillAppear:keyboardDelta:)])
+        {
+            [_delegate ObservingInputAccessoryViewTempKeyboardWillAppear:self keyboardDelta:_keyboardHeight - _previousKeyboardHeight];
+        }
     }
 }
 
 - (void)_keyboardDidShowNotification:(NSNotification*)notification
 {
-    _keyboardState = KeyboardStateShown;
+    if (_keyboardState != KeyboardStateShown) {
+        _keyboardState = KeyboardStateShown;
         
-    [self invalidateIntrinsicContentSize];
+        [self invalidateIntrinsicContentSize];
+    }
 }
 
 - (void)_keyboardWillHideNotification:(NSNotification*)notification
