diff --git a/node_modules/react-16-node-hanging-test-fix/index.js b/node_modules/react-16-node-hanging-test-fix/index.js
index 4a3b342..442eef1 100644
--- a/node_modules/react-16-node-hanging-test-fix/index.js
+++ b/node_modules/react-16-node-hanging-test-fix/index.js
@@ -16,7 +16,7 @@ if (Object.prototype.toString.call(process) !== '[object process]') {
   );
 }
 
-if (semverGt(version, '17.0.1')) {
+if (semverGt(version, '17.0.2')) {
   console.error(
     'The `react-16-node-hanging-test-fix` package is no longer needed ' +
     'with React ' + version + ' and may cause issues. Remove this import.'
