diff --git a/node_modules/react-navigation-header-buttons/src/HeaderButtons.js b/node_modules/react-navigation-header-buttons/src/HeaderButtons.js
index 70ff376..01fba5e 100644
--- a/node_modules/react-navigation-header-buttons/src/HeaderButtons.js
+++ b/node_modules/react-navigation-header-buttons/src/HeaderButtons.js
@@ -144,6 +144,6 @@ const styles = StyleSheet.create({
     }),
   },
   button: {
-    marginHorizontal: 11,
+    marginHorizontal: 6
   },
 });
