diff --git a/node_modules/react-native-math-view/android/build.gradle b/node_modules/react-native-math-view/android/build.gradle
index d960a90..ddf56d7 100644
--- a/node_modules/react-native-math-view/android/build.gradle
+++ b/node_modules/react-native-math-view/android/build.gradle
@@ -31,7 +31,6 @@ repositories {
 buildscript {
     repositories {
         maven { url "https://maven.google.com" }
-        jcenter()
         google()
         maven {
             url 'https://maven.google.com/'
@@ -51,7 +50,6 @@ allprojects {
     repositories {
         mavenLocal()
         maven { url "https://maven.google.com" }
-        jcenter()
         google()
         maven {
             // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
