diff --git a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190129110439.java b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190129110439.java
index 8a85a6b..4065a33 100644
--- a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190129110439.java
+++ b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190129110439.java
@@ -5,7 +5,7 @@ import android.app.Application;
 import android.content.Context;
 import android.content.Intent;
 import android.os.Bundle;
-import android.support.v4.app.NotificationManagerCompat;
+import androidx.core.app.NotificationManagerCompat;
 import android.util.Log;
 
 import com.facebook.react.bridge.Arguments;
diff --git a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190130114725.java b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190130114725.java
index 9f5d49a..b8bbd3c 100644
--- a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190130114725.java
+++ b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190130114725.java
@@ -5,7 +5,7 @@ import android.app.Application;
 import android.content.Context;
 import android.content.Intent;
 import android.os.Bundle;
-import android.support.v4.app.NotificationManagerCompat;
+import androidx.core.app.NotificationManagerCompat;
 import android.util.Log;
 
 import com.facebook.react.bridge.Arguments;
diff --git a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190130114751.java b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190130114751.java
index 3662ab8..9929775 100644
--- a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190130114751.java
+++ b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190130114751.java
@@ -5,7 +5,7 @@ import android.app.Application;
 import android.content.Context;
 import android.content.Intent;
 import android.os.Bundle;
-import android.support.v4.app.NotificationManagerCompat;
+import androidx.core.app.NotificationManagerCompat;
 import android.util.Log;
 
 import com.facebook.react.bridge.Arguments;
diff --git a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190131122445.java b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190131122445.java
index ad8aed2..63dba96 100644
--- a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190131122445.java
+++ b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190131122445.java
@@ -5,7 +5,7 @@ import android.app.Application;
 import android.content.Context;
 import android.content.Intent;
 import android.os.Bundle;
-import android.support.v4.app.NotificationManagerCompat;
+import androidx.core.app.NotificationManagerCompat;
 import android.util.Log;
 
 import com.facebook.react.bridge.ActivityEventListener;
diff --git a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203193225.java b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203193225.java
index 2139835..eee5a4b 100644
--- a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203193225.java
+++ b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203193225.java
@@ -5,7 +5,7 @@ import android.app.Application;
 import android.content.Context;
 import android.content.Intent;
 import android.os.Bundle;
-import android.support.v4.app.NotificationManagerCompat;
+import androidx.core.app.NotificationManagerCompat;
 import android.util.Log;
 
 import com.facebook.react.bridge.ActivityEventListener;
diff --git a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203194421.java b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203194421.java
index 9ee9bc8..614f23f 100644
--- a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203194421.java
+++ b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203194421.java
@@ -5,7 +5,7 @@ import android.app.Application;
 import android.content.Context;
 import android.content.Intent;
 import android.os.Bundle;
-import android.support.v4.app.NotificationManagerCompat;
+import androidx.core.app.NotificationManagerCompat;
 import android.util.Log;
 
 import com.facebook.react.bridge.ActivityEventListener;
diff --git a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203194457.java b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203194457.java
index dd38c7d..056250c 100644
--- a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203194457.java
+++ b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203194457.java
@@ -5,7 +5,7 @@ import android.app.Application;
 import android.content.Context;
 import android.content.Intent;
 import android.os.Bundle;
-import android.support.v4.app.NotificationManagerCompat;
+import androidx.core.app.NotificationManagerCompat;
 import android.util.Log;
 
 import com.facebook.react.bridge.ActivityEventListener;
diff --git a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203194500.java b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203194500.java
index dd38c7d..056250c 100644
--- a/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203194500.java
+++ b/node_modules/react-native-notifications/.history/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule_20190203194500.java
@@ -5,7 +5,7 @@ import android.app.Application;
 import android.content.Context;
 import android.content.Intent;
 import android.os.Bundle;
-import android.support.v4.app.NotificationManagerCompat;
+import androidx.core.app.NotificationManagerCompat;
 import android.util.Log;
 
 import com.facebook.react.bridge.ActivityEventListener;
diff --git a/node_modules/react-native-notifications/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule.java b/node_modules/react-native-notifications/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule.java
index 6588ce0..48f5d3c 100644
--- a/node_modules/react-native-notifications/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule.java
+++ b/node_modules/react-native-notifications/android/src/main/java/com/wix/reactnativenotifications/RNNotificationsModule.java
@@ -5,7 +5,7 @@ import android.app.Application;
 import android.content.Context;
 import android.content.Intent;
 import android.os.Bundle;
-import android.support.v4.app.NotificationManagerCompat;
+import androidx.core.app.NotificationManagerCompat;
 import android.util.Log;
 
 import com.facebook.react.bridge.ActivityEventListener;
diff --git a/node_modules/react-native-notifications/android/src/main/java/com/wix/reactnativenotifications/core/InitialNotificationHolder.java b/node_modules/react-native-notifications/android/src/main/java/com/wix/reactnativenotifications/core/InitialNotificationHolder.java
index 3636f41..ae9ffc4 100644
--- a/node_modules/react-native-notifications/android/src/main/java/com/wix/reactnativenotifications/core/InitialNotificationHolder.java
+++ b/node_modules/react-native-notifications/android/src/main/java/com/wix/reactnativenotifications/core/InitialNotificationHolder.java
@@ -1,6 +1,6 @@
 package com.wix.reactnativenotifications.core;
 
-import android.support.annotation.Nullable;
+import androidx.annotation.Nullable;
 
 import com.wix.reactnativenotifications.core.notification.PushNotificationProps;
 
