diff --git a/node_modules/react-native-external-keyboard/android/src/main/java/com/externalkeyboard/ExternalKeyboardViewPackage.java b/node_modules/react-native-external-keyboard/android/src/main/java/com/externalkeyboard/ExternalKeyboardViewPackage.java
index 98d24e1..cc95665 100644
--- a/node_modules/react-native-external-keyboard/android/src/main/java/com/externalkeyboard/ExternalKeyboardViewPackage.java
+++ b/node_modules/react-native-external-keyboard/android/src/main/java/com/externalkeyboard/ExternalKeyboardViewPackage.java
@@ -5,7 +5,7 @@ import com.externalkeyboard.modules.ExternalKeyboardModule;
 import com.externalkeyboard.views.ExternalKeyboardView.ExternalKeyboardViewManager;
 import com.externalkeyboard.views.KeyboardFocusGroup.KeyboardFocusGroupManager;
 import com.externalkeyboard.views.TextInputFocusWrapper.TextInputFocusWrapperManager;
-import com.facebook.react.BaseReactPackage;
+// import com.facebook.react.BaseReactPackage;
 import com.facebook.react.TurboReactPackage;
 import com.facebook.react.bridge.NativeModule;
 import com.facebook.react.bridge.ReactApplicationContext;
