diff --git a/node_modules/react-native-audio/android/build.gradle b/node_modules/react-native-audio/android/build.gradle
index e43f8bd..49cbbb7 100644
--- a/node_modules/react-native-audio/android/build.gradle
+++ b/node_modules/react-native-audio/android/build.gradle
@@ -26,6 +26,6 @@ def supportVersion = rootProject.hasProperty('supportLibVersion') ? rootProject.
 
 dependencies {
     compile fileTree(include: ['*.jar'], dir: 'libs')
-    compile 'com.android.support:appcompat-v7:${supportVersion}'
+    compile 'androidx.appcompat:appcompat:1.0.0'
     compile 'com.facebook.react:react-native:+'
 }
diff --git a/node_modules/react-native-audio/android/src/main/java/com/rnim/rn/audio/AudioRecorderManager.java b/node_modules/react-native-audio/android/src/main/java/com/rnim/rn/audio/AudioRecorderManager.java
index f6cecf6..e9070b5 100644
--- a/node_modules/react-native-audio/android/src/main/java/com/rnim/rn/audio/AudioRecorderManager.java
+++ b/node_modules/react-native-audio/android/src/main/java/com/rnim/rn/audio/AudioRecorderManager.java
@@ -27,8 +27,8 @@ import android.os.Build;
 import android.os.Environment;
 import android.media.MediaRecorder;
 import android.media.AudioManager;
-import android.support.v4.app.ActivityCompat;
-import android.support.v4.content.ContextCompat;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
 import android.util.Base64;
 import android.util.Log;
 import com.facebook.react.modules.core.DeviceEventManagerModule;
