diff --git a/node_modules/react-native/Libraries/Lists/FlatList.js b/node_modules/react-native/Libraries/Lists/FlatList.js
index f78ca22..d31970a 100644
--- a/node_modules/react-native/Libraries/Lists/FlatList.js
+++ b/node_modules/react-native/Libraries/Lists/FlatList.js
@@ -406,23 +406,23 @@ class FlatList<ItemT> extends React.PureComponent<Props<ItemT>, void> {
 
   constructor(props: Props<ItemT>) {
     super(props);
-    this._checkProps(this.props);
-    if (this.props.viewabilityConfigCallbackPairs) {
+    this._checkProps(props);
+    if (props.viewabilityConfigCallbackPairs) {
       this._virtualizedListPairs =
-        this.props.viewabilityConfigCallbackPairs.map(pair => ({
+        props.viewabilityConfigCallbackPairs.map(pair => ({
           viewabilityConfig: pair.viewabilityConfig,
           onViewableItemsChanged: this._createOnViewableItemsChanged(
             pair.onViewableItemsChanged,
           ),
         }));
-    } else if (this.props.onViewableItemsChanged) {
+    } else if (props.onViewableItemsChanged) {
       this._virtualizedListPairs.push({
         /* $FlowFixMe[incompatible-call] (>=0.63.0 site=react_native_fb) This
          * comment suppresses an error found when Flow v0.63 was deployed. To
          * see the error delete this comment and run Flow. */
-        viewabilityConfig: this.props.viewabilityConfig,
+        viewabilityConfig: props.viewabilityConfig,
         onViewableItemsChanged: this._createOnViewableItemsChanged(
-          this.props.onViewableItemsChanged,
+          props.onViewableItemsChanged,
         ),
       });
     }
@@ -469,7 +469,7 @@ class FlatList<ItemT> extends React.PureComponent<Props<ItemT>, void> {
       onViewableItemsChanged,
       viewabilityConfigCallbackPairs,
     } = props;
-    const numColumns = numColumnsOrDefault(this.props.numColumns);
+    const numColumns = numColumnsOrDefault(props.numColumns);
     invariant(
       !getItem && !getItemCount,
       'FlatList does not support custom data formats.',
