diff --git a/node_modules/react-native-fast-image/ios/FastImage/FFFastImageView.m b/node_modules/react-native-fast-image/ios/FastImage/FFFastImageView.m
index 82188ff..9d47080 100644
--- a/node_modules/react-native-fast-image/ios/FastImage/FFFastImageView.m
+++ b/node_modules/react-native-fast-image/ios/FastImage/FFFastImageView.m
@@ -1,4 +1,5 @@
 #import "FFFastImageView.h"
+#import "UrlCredentials.h"
 
 @interface FFFastImageView()
 
@@ -18,6 +19,9 @@ - (id) init {
     self = [super init];
     self.resizeMode = RCTResizeModeCover;
     self.clipsToBounds = YES;
+
+    [SDWebImageDownloader sharedDownloader].config.urlCredential = [UrlCredentials getUrlCredential];
+
     return self;
 }
 
