diff --git a/node_modules/react-native-math-view/android/build.gradle b/node_modules/react-native-math-view/android/build.gradle
index d960a90..263db9b 100644
--- a/node_modules/react-native-math-view/android/build.gradle
+++ b/node_modules/react-native-math-view/android/build.gradle
@@ -21,7 +21,7 @@ android {
 /*
 repositories {
     mavenCentral()
-    jcenter()
+    mavenCentral()
     google()
     maven {
         url "https://dl.bintray.com/kexanie/maven"
@@ -31,7 +31,7 @@ repositories {
 buildscript {
     repositories {
         maven { url "https://maven.google.com" }
-        jcenter()
+        mavenCentral()
         google()
         maven {
             url 'https://maven.google.com/'
@@ -51,7 +51,7 @@ allprojects {
     repositories {
         mavenLocal()
         maven { url "https://maven.google.com" }
-        jcenter()
+        mavenCentral()
         google()
         maven {
             // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
diff --git a/node_modules/react-native-math-view/android/src/main/java/io/autodidact/rnmathview/SVGShadowNode.java b/node_modules/react-native-math-view/android/src/main/java/io/autodidact/rnmathview/SVGShadowNode.java
index e3719e6..6de29da 100644
--- a/node_modules/react-native-math-view/android/src/main/java/io/autodidact/rnmathview/SVGShadowNode.java
+++ b/node_modules/react-native-math-view/android/src/main/java/io/autodidact/rnmathview/SVGShadowNode.java
@@ -11,7 +11,6 @@ import com.facebook.react.uimanager.NativeViewHierarchyManager;
 import com.facebook.react.uimanager.ReactStylesDiffMap;
 import com.facebook.react.uimanager.UIBlock;
 import com.facebook.react.uimanager.UIManagerModule;
-import com.facebook.react.uimanager.UIManagerModuleListener;
 import com.facebook.react.uimanager.annotations.ReactProp;
 import com.facebook.yoga.YogaMeasureFunction;
 import com.facebook.yoga.YogaMeasureMode;
@@ -79,4 +78,4 @@ public class SVGShadowNode extends LayoutShadowNode implements YogaMeasureFuncti
         int measuredHeight = View.resolveSize((int) svgAttributes.height, heightSpec);
         return YogaMeasureOutput.make(measuredWidth, measuredHeight);
     }
-}
\ No newline at end of file
+}
