diff --git a/node_modules/react-native-document-picker/ios/RNDocumentPicker/RNDocumentPicker.m b/node_modules/react-native-document-picker/ios/RNDocumentPicker/RNDocumentPicker.m
index 1cb8b4b..43f2cc6 100644
--- a/node_modules/react-native-document-picker/ios/RNDocumentPicker/RNDocumentPicker.m
+++ b/node_modules/react-native-document-picker/ios/RNDocumentPicker/RNDocumentPicker.m
@@ -10,6 +10,8 @@
 #import "RCTBridge.h"
 #endif
 
+#import <React/RCTLog.h>
+
 static NSString *const E_DOCUMENT_PICKER_CANCELED = @"DOCUMENT_PICKER_CANCELED";
 static NSString *const E_INVALID_DATA_RETURNED = @"INVALID_DATA_RETURNED";
 
@@ -58,7 +60,7 @@ - (dispatch_queue_t)methodQueue
                   rejecter:(RCTPromiseRejectBlock)reject)
 {
     NSArray *allowedUTIs = [RCTConvert NSArray:options[OPTION_TYPE]];
-    UIDocumentPickerViewController *documentPicker = [[UIDocumentPickerViewController alloc] initWithDocumentTypes:(NSArray *)allowedUTIs inMode:UIDocumentPickerModeImport];
+    UIDocumentPickerViewController *documentPicker = [[UIDocumentPickerViewController alloc] initWithDocumentTypes:(NSArray *)allowedUTIs inMode:UIDocumentPickerModeOpen];
     
     [composeResolvers addObject:resolve];
     [composeRejecters addObject:reject];
@@ -92,7 +94,6 @@ - (NSMutableDictionary *)getMetadataForUrl:(NSURL *)url error:(NSError **)error
     [coordinator coordinateReadingItemAtURL:url options:NSFileCoordinatorReadingResolvesSymbolicLink error:&fileError byAccessor:^(NSURL *newURL) {
         
         if (!fileError) {
-            [result setValue:newURL.absoluteString forKey:FIELD_URI];
             [result setValue:[newURL lastPathComponent] forKey:FIELD_NAME];
             
             NSError *attributesError = nil;
@@ -112,6 +113,12 @@ - (NSMutableDictionary *)getMetadataForUrl:(NSURL *)url error:(NSError **)error
                 NSString *mimeTypeString = (__bridge_transfer NSString *)mimeType;
                 [result setValue:mimeTypeString forKey:FIELD_TYPE];
             }
+
+            NSString *path = [NSTemporaryDirectory() stringByAppendingPathComponent:newURL.lastPathComponent];
+            NSData *data = [NSData dataWithContentsOfURL:newURL];
+            [data writeToFile:path atomically:YES];
+
+            [result setValue:[@"file://" stringByAppendingString:path] forKey:FIELD_URI];
         }
     }];
     
@@ -127,7 +134,7 @@ - (NSMutableDictionary *)getMetadataForUrl:(NSURL *)url error:(NSError **)error
 
 - (void)documentPicker:(UIDocumentPickerViewController *)controller didPickDocumentAtURL:(NSURL *)url
 {
-    if (controller.documentPickerMode == UIDocumentPickerModeImport) {
+    if (controller.documentPickerMode == UIDocumentPickerModeOpen) {
         RCTPromiseResolveBlock resolve = [composeResolvers lastObject];
         RCTPromiseRejectBlock reject = [composeRejecters lastObject];
         [composeResolvers removeLastObject];
@@ -146,7 +153,7 @@ - (void)documentPicker:(UIDocumentPickerViewController *)controller didPickDocum
 
 - (void)documentPicker:(UIDocumentPickerViewController *)controller didPickDocumentsAtURLs:(NSArray<NSURL *> *)urls
 {
-    if (controller.documentPickerMode == UIDocumentPickerModeImport) {
+    if (controller.documentPickerMode == UIDocumentPickerModeOpen) {
         RCTPromiseResolveBlock resolve = [composeResolvers lastObject];
         RCTPromiseRejectBlock reject = [composeRejecters lastObject];
         [composeResolvers removeLastObject];
