diff --git a/node_modules/react-native-webview/android/.project b/node_modules/react-native-webview/android/.project
new file mode 100644
index 0000000..122755b
--- /dev/null
+++ b/node_modules/react-native-webview/android/.project
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>react-native-webview</name>
+	<comment>Project react-native-webview created by Buildship.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+	<filteredResources>
+		<filter>
+			<id>1760118463063</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|\.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>
diff --git a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java
index 5932fc1..1ce86e8 100644
--- a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java
+++ b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java
@@ -223,6 +223,7 @@ public class RNCWebView extends WebView implements LifecycleEventListener {
     public void setWebViewClient(WebViewClient client) {
         super.setWebViewClient(client);
         if (client instanceof RNCWebViewClient) {
+            ((RNCWebViewClient) client).setReactContext(this.getReactApplicationContext());
             mRNCWebViewClient = (RNCWebViewClient) client;
             mRNCWebViewClient.setProgressChangedFilter(progressChangedFilter);
         }
diff --git a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewClient.java b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewClient.java
index 251939e..763cc50 100644
--- a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewClient.java
+++ b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewClient.java
@@ -34,6 +34,11 @@ import com.reactnativecommunity.webview.events.TopShouldStartLoadWithRequestEven
 import android.webkit.CookieManager;
 import android.webkit.CookieSyncManager;
 
+import java.security.cert.X509Certificate;
+import java.security.PrivateKey;
+import android.webkit.ClientCertRequest;
+import android.os.AsyncTask;
+import android.security.KeyChain;
 import java.util.concurrent.atomic.AtomicReference;
 
 public class RNCWebViewClient extends WebViewClient {
@@ -45,6 +50,9 @@ public class RNCWebViewClient extends WebViewClient {
     protected @Nullable String ignoreErrFailedForThisURL = null;
     protected @Nullable RNCBasicAuthCredential basicAuthCredential = null;
 
+    private static String certificateAlias = null;
+    private ReactContext reactContext;
+
     public void setIgnoreErrFailedForThisURL(@Nullable String url) {
         ignoreErrFailedForThisURL = url;
     }
@@ -53,6 +61,53 @@ public class RNCWebViewClient extends WebViewClient {
         basicAuthCredential = credential;
     }
 
+    public static void setCertificateAlias(String alias) {
+        certificateAlias = alias;
+    }
+
+    public void setReactContext(ReactContext reactContext) {
+        this.reactContext = reactContext;
+    }
+
+    @Override
+    public void onReceivedClientCertRequest(WebView view, ClientCertRequest request) {
+        class SslStuff {
+            PrivateKey privKey;
+            X509Certificate[] certChain;
+
+            public SslStuff(PrivateKey privKey, X509Certificate[] certChain) {
+                this.privKey = privKey;
+                this.certChain = certChain;
+            }
+        }
+
+        if (certificateAlias != null && reactContext != null) {
+            AsyncTask<Void, Void, SslStuff> task = new AsyncTask<Void, Void, SslStuff>() {
+                @Override
+                protected SslStuff doInBackground(Void... params) {
+                    try {
+                        PrivateKey privKey = KeyChain.getPrivateKey(reactContext, certificateAlias);
+                        X509Certificate[] certChain = KeyChain.getCertificateChain(reactContext, certificateAlias);
+
+                        return new SslStuff(privKey, certChain);
+                    } catch (Exception e) {
+                        return null;
+                    }
+                }
+
+                @Override
+                protected void onPostExecute(SslStuff sslStuff) {
+                    if (sslStuff != null) {
+                        request.proceed(sslStuff.privKey, sslStuff.certChain);
+                    }
+                }
+            };
+            task.execute();
+        } else {
+            super.onReceivedClientCertRequest(view, request);
+        }
+    }
+
     @Override
     public void onPageFinished(WebView webView, String url) {
         super.onPageFinished(webView, url);
diff --git a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModuleImpl.java b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModuleImpl.java
index 951749c..12a0bd2 100644
--- a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModuleImpl.java
+++ b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModuleImpl.java
@@ -551,4 +551,8 @@ public class RNCWebViewModuleImpl implements ActivityEventListener {
         }
         return (PermissionAwareActivity) activity;
     }
+    
+    public static void setCertificateAlias(String alias) {
+        RNCWebViewClient.setCertificateAlias(alias);
+    }
 }
\ No newline at end of file
diff --git a/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewManager.java b/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewManager.java
index e20ae03..01887a6 100644
--- a/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewManager.java
+++ b/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewManager.java
@@ -44,6 +44,10 @@ public class RNCWebViewManager extends ViewGroupManager<RNCWebViewWrapper>
         mRNCWebViewManagerImpl = new RNCWebViewManagerImpl(true);
     }
 
+    public static void setCertificateAlias(String alias) {
+        RNCWebViewModule.setCertificateAlias(alias);
+    }
+
     @Nullable
     @Override
     protected ViewManagerDelegate<RNCWebViewWrapper> getDelegate() {
diff --git a/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewModule.java b/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewModule.java
index 4df0aae..867aa73 100644
--- a/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewModule.java
+++ b/node_modules/react-native-webview/android/src/newarch/com/reactnativecommunity/webview/RNCWebViewModule.java
@@ -49,6 +49,10 @@ public class RNCWebViewModule extends NativeRNCWebViewModuleSpec {
         return mRNCWebViewModuleImpl.grantFileDownloaderPermissions(downloadingMessage, lackPermissionToDownloadMessage);
     }
 
+    public static void setCertificateAlias(String alias) {
+        RNCWebViewModuleImpl.setCertificateAlias(alias);
+    }
+
     @NonNull
     @Override
     public String getName() {
diff --git a/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewManager.java b/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewManager.java
index 44d3378..3972014 100644
--- a/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewManager.java
+++ b/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewManager.java
@@ -31,6 +31,10 @@ public class RNCWebViewManager extends ViewGroupManager<RNCWebViewWrapper> {
         mRNCWebViewManagerImpl = new RNCWebViewManagerImpl();
     }
 
+    public static void setCertificateAlias(String alias) {
+        RNCWebViewModule.setCertificateAlias(alias);
+    }
+
     @Override
     public String getName() {
         return RNCWebViewManagerImpl.NAME;
diff --git a/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewModule.java b/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewModule.java
index 526cc16..8f3318d 100644
--- a/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewModule.java
+++ b/node_modules/react-native-webview/android/src/oldarch/com/reactnativecommunity/webview/RNCWebViewModule.java
@@ -51,6 +51,10 @@ public class RNCWebViewModule extends ReactContextBaseJavaModule {
         return mRNCWebViewModuleImpl.grantFileDownloaderPermissions(downloadingMessage, lackPermissionToDownloadMessage);
     }
 
+    public static void setCertificateAlias(String alias) {
+        RNCWebViewModuleImpl.setCertificateAlias(alias);
+    }
+
     @NonNull
     @Override
     public String getName() {
