diff --git a/node_modules/remove-markdown/index.js b/node_modules/remove-markdown/index.js
index 7f53a9c..dea8149 100644
--- a/node_modules/remove-markdown/index.js
+++ b/node_modules/remove-markdown/index.js
@@ -7,6 +7,9 @@ module.exports = function(md, options) {
 
   var output = md || '';
 
+  // Remove surrounding asterisks (*) and any spaces around the inner content
+  output = output.replace(/\*\s*(.*?)\s*\*/g, '*$1*');
+
   // Remove horizontal rules (stripListHeaders conflict with this rule, which is why it has been moved to the top)
   output = output.replace(/^(-\s*?|\*\s*?|_\s*?){3,}\s*$/gm, '');
 
