diff --git a/node_modules/rn-root-view/android/src/main/java/chat/rocket/rootview/RootViewModule.java b/node_modules/rn-root-view/android/src/main/java/chat/rocket/rootview/RootViewModule.java
index 854ff74..823aa38 100644
--- a/node_modules/rn-root-view/android/src/main/java/chat/rocket/rootview/RootViewModule.java
+++ b/node_modules/rn-root-view/android/src/main/java/chat/rocket/rootview/RootViewModule.java
@@ -1,28 +1,43 @@
 package chat.rocket.rootview;
 
-import androidx.annotation.NonNull;
+import android.app.Activity;
+import android.view.View;
+import android.graphics.Color;
 
-import com.facebook.react.ReactPackage;
-import com.facebook.react.bridge.NativeModule;
 import com.facebook.react.bridge.ReactApplicationContext;
-import com.facebook.react.uimanager.ViewManager;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.Callback;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
+public class RootViewModule extends ReactContextBaseJavaModule {
 
-public class RootViewPackage implements ReactPackage {
-    @NonNull
-    @Override
-    public List<NativeModule> createNativeModules(@NonNull ReactApplicationContext reactContext) {
-        List<NativeModule> modules = new ArrayList<>();
-        modules.add(new RootViewModule(reactContext));
-        return modules;
+    private final ReactApplicationContext reactContext;
+
+    public RootViewModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+        this.reactContext = reactContext;
     }
 
-    @NonNull
     @Override
-    public List<ViewManager> createViewManagers(@NonNull ReactApplicationContext reactContext) {
-        return Collections.emptyList();
+    public String getName() {
+        return "RootView";
+    }
+
+    @ReactMethod
+    public void setColor(final String color) {
+        final Activity activity = getCurrentActivity();
+
+        if (activity == null) {
+            return;
+        }
+
+        activity.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                View rootView = activity.getWindow().getDecorView();
+                int parsedColor = Color.parseColor(color);
+                rootView.getRootView().setBackgroundColor(parsedColor);
+            }
+        });
     }
 }
\ No newline at end of file