diff --git a/node_modules/rn-user-defaults/android/src/main/java/chat/rocket/userdefaults/RNUserDefaultsModule.java b/node_modules/rn-user-defaults/android/src/main/java/chat/rocket/userdefaults/RNUserDefaultsModule.java
index 9032629..7b228d2 100644
--- a/node_modules/rn-user-defaults/android/src/main/java/chat/rocket/userdefaults/RNUserDefaultsModule.java
+++ b/node_modules/rn-user-defaults/android/src/main/java/chat/rocket/userdefaults/RNUserDefaultsModule.java
@@ -19,14 +19,16 @@ import com.facebook.react.bridge.WritableMap;
 import java.util.Map;
 
 public class RNUserDefaultsModule extends ReactContextBaseJavaModule {
-  private String preferencesName = "react-native";
-  private String contextName = null;
+  private static String preferencesName = "react-native";
+  private static String contextName = null;
+  private static Context context;
 
   private final ReactApplicationContext reactContext;
 
   public RNUserDefaultsModule(ReactApplicationContext reactContext) {
     super(reactContext);
     this.reactContext = reactContext;
+    this.context = getReactApplicationContext();
   }
 
   @Override
@@ -79,8 +81,7 @@ public class RNUserDefaultsModule extends ReactContextBaseJavaModule {
     promise.resolve(preferencesName);
   }
 
-  private SharedPreferences getPreferences() {
-    Context context = getReactApplicationContext();
+  public static SharedPreferences getPreferences() {
     if (contextName != null) {
       try {
         context = context.createPackageContext(contextName, Context.CONTEXT_INCLUDE_CODE);
@@ -92,6 +93,7 @@ public class RNUserDefaultsModule extends ReactContextBaseJavaModule {
     }
     return context.getSharedPreferences(preferencesName, Context.MODE_PRIVATE);
   }
+
   private SharedPreferences.Editor getEditor() {
     return getPreferences().edit();
   }
